UnityProject/Assets/Scripts/Systems/Spells/Wizard/TeleportSpell.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections;
+using UnityEngine;
+using Mirror;
+
+namespace Systems.Spells.Wizard
+{
+	public class TeleportSpell : NetworkBehaviour
+	{
+		private const float TELEPORT_TRAVEL_TIME = 0;
+		private const float TELEPORT_ANIMATE_TIME = 0.8f;
+		private const float TELEPORT_ANIMATE_X_SCALE = 0.4f;
+		private const float TELEPORT_ANIMATE_Y_SCALE = 1.1f;
+		private const float TELEPORT_ANIMATE_HEIGHT = 1.5f;
+
+		public bool IsBusy { get; private set; } = false;
+
+		// We sync the teleporting player so we can play animations locally.
+		[SyncVar(hook = nameof(SyncPlayer))]
+		private GameObject teleportingPlayer;
+
+		private Transform playerSprite;
+
+		/// <summary>
+		/// When set true, plays teleport begin animation on the client. False: plays teleport end animation.
+		/// </summary>
+		[SyncVar(hook = nameof(SyncAnimation))]
+		private bool syncAnimation = false;
+
+		public void ServerTeleportWizard(GameObject playerToTeleport, Vector3Int toWorldPos)
+		{
+			teleportingPlayer = playerToTeleport;
+
+			StartCoroutine(RunTeleportSequence(toWorldPos));
+		}
+
+		private IEnumerator RunTeleportSequence(Vector3Int toWorldPos)
+		{
+			IsBusy = true;
+			syncAnimation = true;
+			yield return WaitFor.Seconds(TELEPORT_ANIMATE_TIME + TELEPORT_TRAVEL_TIME);
+
+			teleportingPlayer.GetComponent<PlayerSync>().SetPosition(toWorldPos, true);
+
+			syncAnimation = false;
+			yield return WaitFor.Seconds(TELEPORT_ANIMATE_TIME);
+			IsBusy = false;
+		}
+
+		private void SyncPlayer(GameObject oldPlayer, GameObject newPlayer)
+		{
+			teleportingPlayer = newPlayer;
+			playerSprite = teleportingPlayer.transform.Find("Sprites");
+
+			if (playerSprite == null)
+			{
+				Logger.LogError($"Couldn't find child GameObject 'Sprites' on {teleportingPlayer}. Has the hierarchy changed?");
+			}
+		}
+
+		private void SyncAnimation(bool oldState, bool newState)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'oldState'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6316546)